File: app/src/main/java/org/nekoiceteam/nekoicebot/Bot.java
package org.nekoiceteam.nekoicebot;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import javax.security.auth.login.LoginException;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.utils.Compression;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.OnlineStatus;
import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.requests.GatewayIntent;
import net.dv8tion.jda.api.utils.ChunkingFilter;
import net.dv8tion.jda.api.utils.MemberCachePolicy;
import org.nekoiceteam.nekoicebot.events.Join;
import org.nekoiceteam.nekoicebot.utils.Logger;

public class Bot {
        
        JDABuilder Builder;
        JDA Client;
        Logger logger;
        
        
	public Bot(String[] args) throws NullPointerException, LoginException, IOException, IllegalArgumentException, InterruptedException {
            
            logger = new Logger("Log");
            logger.info("Initializing bot!");
            
            List<String> list = Files.readAllLines(Paths.get("config.txt"));
            String token = list.get(0);
            
		JDABuilder builder = JDABuilder.createDefault(token);
		
		builder.setBulkDeleteSplittingEnabled(false);
		builder.setCompression(Compression.ZLIB);
		builder.setActivity(Activity.playing("With Your Mom"));
		builder.setStatus(OnlineStatus.IDLE);
		builder.setChunkingFilter(ChunkingFilter.ALL);
        	builder.setMemberCachePolicy(MemberCachePolicy.ALL);
                builder.enableIntents(GatewayIntent.GUILD_MEMBERS, GatewayIntent.GUILD_MESSAGES);
                builder.addEventListeners(new MessageListener(), new Join());
                Client = builder.build();
                Client.awaitReady();
                logger.info("Bot is ready!");
            }
}

File: app/src/main/kotlin/org/nekoiceteam/nekoicebot/bot/Bot.kt
package org.nekoiceteam.nekoicebot.bot

import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.JDABuilder
import net.dv8tion.jda.api.OnlineStatus
import net.dv8tion.jda.api.entities.Activity
import net.dv8tion.jda.api.requests.GatewayIntent
import net.dv8tion.jda.api.utils.ChunkingFilter
import net.dv8tion.jda.api.utils.Compression
import net.dv8tion.jda.api.utils.MemberCachePolicy
import org.nekoiceteam.nekoicebot.bot.commands.utils.test
import org.nekoiceteam.nekoicebot.bot.events.Join
import org.nekoiceteam.nekoicebot.bot.utils.Logger
import java.nio.file.Files
import java.nio.file.Paths

class Bot(args: Array<String?>?) {
    var Builder: JDABuilder? = null
    var Client: JDA
    var logger: Logger

    init {
        logger = Logger("latest")
        logger.info("Initializing bot!")
        val list = Files.readAllLines(Paths.get("config.txt"))
        val token = list[0]
        val builder = JDABuilder.createDefault(token)
        builder.setBulkDeleteSplittingEnabled(false)
        builder.setCompression(Compression.ZLIB)
        builder.setActivity(Activity.playing("With Your Mom"))
        builder.setStatus(OnlineStatus.IDLE)
        builder.setChunkingFilter(ChunkingFilter.ALL)
        builder.setMemberCachePolicy(MemberCachePolicy.ALL)
        builder.enableIntents(GatewayIntent.GUILD_MEMBERS, GatewayIntent.GUILD_MESSAGES)
        builder.addEventListeners(Join(), test())
        Client = builder.build()
        Client.awaitReady()
        logger.info("Bot is ready!")
        Client.upsertCommand("test", "testing slash command").queue()
    }
}